support-services/src/main/scala/com/gu/support/catalog/CatalogService.scala

```diff
@@ -14,6 +14,34 @@ object CatalogService {
 }
 
 class CatalogService(val environment: TouchPointEnvironment, jsonProvider: CatalogJsonProvider) extends LazyLogging {
+
+  private def getRatePlanId(billingPeriod: BillingPeriod, fulfilmentOptions: FulfilmentOptions)
+    = getProductRatePlan(environment, billingPeriod,fulfilmentOptions, NoProductOptions).map(_.id).getOrElse("")
+
+  private def fetchQuarterlyPrice(quarterlyId: ProductRatePlanId, sixWeeklyPriceList: Pricelist, catalogPrices: List[Pricelist]) = {
+    Pricelist(sixWeeklyPriceList.productRatePlanId,
+      catalogPrices.find(_.productRatePlanId == quarterlyId).map(_.prices).getOrElse(sixWeeklyPriceList.prices)
+    )
+  }
+
+  def adjustSixWeeklyPriceList(c: Catalog) = {
```

Comment:
All of this is just to map between Zuora's representation of a 6 for 6 subs which is made up of 2 separate rate plans and the representation we use on the site which is a quarterly subscription with an introductory promotion